ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/OfferingToStateMapper.kt

```diff
@@ -45,14 +50,23 @@ private fun PaywallData.validate(): Result<PaywallTemplate> {
 
     val invalidVariablesError = localizedConfiguration.validateVariables()
     if (invalidVariablesError != null) {
-        return Result.failure(invalidVariablesError)
+        return Result.failure<PaywallTemplate>(invalidVariablesError).also {
+            val joinedUnrecognizedVariables = invalidVariablesError.unrecognizedVariables.joinToString()
+            Logger.d(PaywallValidationErrorStrings.INVALID_VARIABLES.format(joinedUnrecognizedVariables))
```

Comment:
This is kind duplicated in all of these `return`s.
Should we log it in a single line in `PaywallViewModel.calculateState` instead, when `PaywallValidationError` isn't `null`?